src/test/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessageTest.java

```diff
@@ -227,6 +228,23 @@ private static LocalizedMessage createSampleLocalizedMessage() {
                 "empty.statement", EMPTY_OBJECT_ARRAY, "module", LocalizedMessage.class, null);
     }
 
+    @Test
+    public void testCompareToWithDifferentModuleId() {
+        final LocalizedMessage message1 = new LocalizedMessage(0,
+                "com.puppycrawl.tools.checkstyle.checks.coding.messages", "empty.statement",
+                EMPTY_OBJECT_ARRAY, "module1", LocalizedMessage.class, null);
+        final LocalizedMessage message2 = new LocalizedMessage(0,
+                "com.puppycrawl.tools.checkstyle.checks.coding.messages", "empty.statement",
+                EMPTY_OBJECT_ARRAY, null, LocalizedMessage.class, null);
+        final LocalizedMessage message3 = new LocalizedMessage(0,
```

Comment:
It's hard to follow what is the difference between these 3.
Rename `message3` to `message2` and rename current `message2` to `messageNull`.